### PR TITLE
Add ability to link test results with issues and tms.

### PR DESCRIPTION
**Pull request by damonpam**
### Context
Currently, with the new version, it is not possible to link test results with issue tracker tools, like JIRA or XRAY

#### Checklist
- [x] [Sign Allure CLA][cla]
- [x] Provide unit tests

### CODE REVIEW

File: packages/allure-cucumberjs/README.md [hunk 1]
```diff
@@ -11,23 +11,33 @@ export default class Reporter extends CucumberJSAllureFormatter {
...
       {
         labels: {
-          issue: [/@bug_(.*)/],
-          epic: [/@feature:(.*)/]
-        }
+          epic: [/@feature:(.*)/],
+          issue: [/@issue:(.*)/],
+          tms: [/@tms:(.*)/],
+        },
+        issuesTrackerUrlPattern: "http://localhost:8080/issue/%s",
```
Comment (baev) [inline packages/allure-cucumberjs/README.md:21]: allure 2 supports different link types. So can we make it like for labels, where you have links configuration section with link type as key and pattern as value? 

```json
links: {
   issue: "http://localhost:8080/issue/%s",
   tms: "http://localhost:8080/tms/%s",
   custom: "http://localhost:8080/custom/%s"
}
```

File: packages/allure-js-commons/src/AllureTest.ts [hunk 2]
```diff
@@ -37,4 +37,12 @@ export class AllureTest extends ExecutableItemWrapper {
   addLink(url: string, name?: string, type?: string): void {
     this.testResult.links.push({ name, url, type });
   }
+
+  addIssue(url: string, name: string) {
+    this.addLink(url, name, LinkType.ISSUE);
+  }
+
+  addTMS(url: string, name: string) {
```
Comment (baev) [inline packages/allure-js-commons/src/AllureTest.ts:45]: addTms

**Review by baev — CHANGES_REQUESTED** [review]

**Review by just-boris — APPROVED** [review]
Not super familiar with Cucumber, but no concerns on my side

File: packages/allure-js-commons/src/AllureTest.ts (hunk 2 above)
Comment (baev) [reply packages/allure-js-commons/src/AllureTest.ts:45]: or even better `addTmsLink` | `addIssueLink`

Comment (damonpam) [reply packages/allure-js-commons/src/AllureTest.ts:45]: done

**Comment by damonpam** [comment]
@baev changes done

File: packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts [hunk 3]
```diff
@@ -172,67 +199,134 @@ export class CucumberJSAllureFormatter extends Formatter {
...
       }
+
+      for (const link in this.links) {
+        if (!this.links[link]) {
+          continue;
+        }
+        for (const reg of this.links[link]) {
+          const match = tag.name.match(reg);
+          if (match != null && match.length > 1) {
+            if (link === LinkType.ISSUE && this.issuesTrackerUrlPattern) {
```
Comment (baev) [inline packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts:220]: How could I create a link with custom type? Maybe add url template to `links` configuration section? Something like

```js
links: {
  issue: {
     pattern: [/@issue:(.*)/]
     urlTemplate: "http://localhost:8080/issue/%s"
  },
  tms: {
     pattern: [/@tms:(.*)/]
     urlTemplate: "http://localhost:8080/tms/%s"
  }
},
```

**Review by baev — CHANGES_REQUESTED** [review]

**Comment by maga-polito** [comment]
Hi, is it possible to have this available soon?

File: packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts (hunk 3 above)
Comment (damonpam) [reply packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts]: ok!

Comment (damonpam) [reply packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts]: @baev Sorry for the delay, I've been very busy these last few days... check the new code, please.

**Comment by damonpam** [comment]
Now the links (only issue and tms) are configurable... Updated the documentation according to that. Thanks in advance!

**Review by baev — APPROVED** [review]

**Comment by damonpam** [comment]
Who can merge this? @baev

**Comment by damonpam** [comment]
@sseliverstov how do we release a new version of the npm package?

**Pull request merged**